tests/functional/u/use/used_before_assignment_issue2615.py

```diff
@@ -4,6 +4,42 @@ def main():
     try:
         res = 1 / 0
         res = 42
+        if main():
+            res = None
+        with open(__file__, encoding="utf-8") as opened_file:
+            res = opened_file.readlines()
     except ZeroDivisionError:
         print(res)  # [used-before-assignment]
     print(res)
+
+
+def nested_except_blocks():
+    """Don't confuse except blocks with each other."""
+    try:
+        res = 1 / 0
+        res = 42
+        if main():
+            res = None
+        with open(__file__, encoding="utf-8") as opened_file:
+            res = opened_file.readlines()
+    except ZeroDivisionError:
+        try:
+            more_bad_division = 1 / 0
+        except ZeroDivisionError:
+            print(more_bad_division)  # [used-before-assignment]
+            print(res)  # don't emit: too nested to make confident inferences
```

Comment:
This seems like a good follow-up issue? Or is this too complicated to even try and tackle?

Reply:
I might be able to solve it on this branch, let me see after similar changes in https://github.com/PyCQA/pylint/pull/5583/commits/09a725136a3f5582d1ab01124e1888b48d4173fb

Reply:
Actually didn't end up relying on the util function in the other PR. Done in 18dd196

Reply:
... but fixing the potential regression that the CI suite found _did_ end up relying on the util function introduced in the other PR.